domains/certificates/Query.API/API/DataSyncSyncer/Service/DataSync.cs

```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Security.Claims;
+using System.Text;
+using System.Threading.Tasks;
+using CertificateEvents.Primitives;
+using IdentityModel.Client;
+using Microsoft.Extensions.Logging;
+using Microsoft.IdentityModel.Tokens;
+
+namespace API.DataSyncSyncer.Service.Datasync;
+
+public class DataSync : IDataSync
+{
+    private readonly HttpClient httpClient;
+    private readonly ILogger<DataSync> logger;
+
+    public DataSync(HttpClient httpClient, ILogger<DataSync> logger)
+    {
+        this.logger = logger;
+        this.httpClient = httpClient;
+    }
+
+    public async Task<List<DataSyncDto>> GetMeasurement(string gsrn, Period period, string meteringPointOwner)
+    {
+        logger.LogInformation(
+            "Fetching data in period from {from} to: {to}", period.DateFrom, period.DateTo
+        );
+
+        var url = $"measurements?gsrn={gsrn}&dateFrom={period.DateFrom}&dateTo={period.DateTo}";
+
+        httpClient.SetBearerToken(GenerateToken(meteringPointOwner));
+
+        var response = await httpClient.GetAsync(url);
+
+        return await response.Content.ReadFromJsonAsync<List<DataSyncDto>>()
+               ?? throw new Exception($"Fetch of measurements failed, base: {httpClient.BaseAddress} url: {url}");
```

Comment:
If there is a failure, then this will throw an exception. I do not see something that will catch this exception. So a single exception will bring down the entire DataSyncSyncerWorker.
I suggest that we here will not throw an exception if DataSync is not responding, but will return either an empty result or a type indicating that something went wrong.